controllers/kustomization_controller.go

```diff
@@ -490,6 +490,10 @@ func (r *KustomizationReconciler) checkDependencies(kustomization kustomizev1.Ku
 		if !apimeta.IsStatusConditionTrue(k.Status.Conditions, meta.ReadyCondition) {
 			return fmt.Errorf("dependency '%s' is not ready", dName)
 		}
+		
+		if len(k.Status.Conditions) == 0 || k.Spec.SourceRef.Name == kustomization.Spec.SourceRef.Name && k.Spec.SourceRef.Namespace == kustomization.Spec.SourceRef.Namespace && source.GetArtifact().Revision != k.Status.LastAppliedRevision {
```

Comment:
There is no need for `len(k.Status.Conditions) == 0` this is evaluated above. Another issue is that we only check the source name and namespace, but we really need to check the kind as well, as a GitRepository can have the same name as a Bucket.

Reply:
Fair enough i already thought about other specs to check but forgot the kind.

Reply:
I added the kind Check and remove the duplicated Conditions Check